Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp

```diff
@@ -123,8 +123,10 @@ void RemoteImageBufferSet::ensureBufferForDisplay(const ImageBufferSetPrepareBuf
     bool needsFullDisplay = false;
     if (m_frontBuffer) {
         auto previousState = m_frontBuffer->setNonVolatile();
-        if (previousState == SetNonVolatileResult::Empty)
+        if (previousState == SetNonVolatileResult::Empty) {
             needsFullDisplay = true;
+            inputData.dirtyRegion = IntRect { { }, expandedIntSize(m_logicalSize) };
```

Comment:
slightly odd that needsFullDisplay doesn't imply this

Reply:
The joys of the current code is that we have to iterate all the layers doing the swapping first, so that we can get the buffer handles sent back the web process as soon as possible (since it'll block waiting for that message at the end of painting).

Then once we've done that, we iterate a second time, initiating all the copy-forward operations.

Since there's two separate functions run on the `RemoteImageBufferSet`, `needsFullDisplay` isn't visible to the latter. Expanding the dirty region is the easiest way to share that information (that I could think of).